rotkehlchen/chain/ethereum/modules/defisaver/decoder.py

```diff
@@ -64,10 +145,69 @@ def _decode_substorage_action(self, context: DecoderContext) -> DecodingOutput:
 
         return DEFAULT_DECODING_OUTPUT
 
+    def _decode_defisaver_events(self, context: DecoderContext) -> DecodingOutput:
+        """
+        Decodes DeFi Saver events from the given context.
+
+        This method specifically matches the counterparty
+        so to trigger the post-decoding rules.
+        """
+        # We want only to remap the ERC20 transfers, so
+        # we just match the counterparty for triggering the post-decoding rules
+        return DecodingOutput(matched_counterparty=CPT_DEFISAVER)
+
+    def _decode_flashloan(
+            self,
+            decoded_events: list['EvmEvent'],
+            transaction: 'EvmTransaction',
+            all_logs: list['EvmTxReceiptLog'],
+    ) -> list['EvmEvent']:
+        """
+        Decodes flashloan events from the given decoded events.
+
+        This method identifies flashloan ERC20 transfer events originating from DefiSaver flashloan
+        contracts and sets the appropriate event subtype.
+        """
+        for event in decoded_events:
+            if event.address in self.flashloan_addresses and self.base.is_tracked(event.location_label):  # noqa: E501
```

Comment:
instead of post decoding can't you do it in the normal `address -> decoding_rule`? Asking since you check for a known subset of addresses

![image](https://github.com/user-attachments/assets/7403ce5e-9ba1-4cdf-bba3-b4ff50452b1c)

also there is an action item but is it possible to extract that it is a flashloan?

Reply:
Well, I thought that post decoding was enough since ERC20 transfers were already decoded. If we do in a decoding_rule, erc20 transfer decoding of that event would be skipped? And, if yes, it is worh replacing a decoding of a transfer that is already performed in ERC20 transfer decoding? Those were my questions when I did it.

Reply:
And the ActionEvent is not existing for all the contracts, and I haven't the ABI for the data section. Since flashloan is a transfer, I thought that post decoding of them was enough, while using the FLActionEvents (when existing) or the DFSLogger events for matching the transaction with DFS counterparty. Then post decoding looks for the flashloan transfers. 

Those were my ideas at least.